### ISSUE TITLE
ValueError: not enough values to unpack (expected 4, got 0)

### ISSUE THREAD

**Issue by imai-lm** [dmlc/nnvm#204]
This error below occurs frequently in my environment. Does anyone understand what is the problem?
```
LM-MacBook-27:nnvm-test2 imai$ python nnvm-onnx-import.py > graph_nnvm.txt 
Traceback (most recent call last):
  File "nnvm-onnx-import.py", line 45, in <module>
    graph, lib, params = nnvm.compiler.build(sym, target, shape_dict, params=params, dtype="float32")
  File "/Users/imai/.local/lib/python3.5/site-packages/nnvm-0.8.0-py3.5.egg/nnvm/compiler/build_module.py", line 237, in build
    graph = graph.apply("GraphFusePartition").apply("GraphFuseCompile")
  File "/Users/imai/.local/lib/python3.5/site-packages/nnvm-0.8.0-py3.5.egg/nnvm/graph.py", line 234, in apply
    check_call(_LIB.NNGraphApplyPasses(self.handle, npass, cpass, ctypes.byref(ghandle)))
  File "/Users/imai/.local/lib/python3.5/site-packages/nnvm-0.8.0-py3.5.egg/nnvm/_base.py", line 72, in check_call
    raise NNVMError(py_str(_LIB.NNGetLastError()))
nnvm._base.NNVMError: TVMCall CFunc Error:
Traceback (most recent call last):
  File "/Users/imai/.pyenv/versions/3.5.3/lib/python3.5/site-packages/tvm-0.1.0-py3.5-macosx-10.10-x86_64.egg/tvm/_ffi/_ctypes/function.py", line 54, in cfun
    rv = local_pyfunc(*pyargs)
  File "/Users/imai/.local/lib/python3.5/site-packages/nnvm-0.8.0-py3.5.egg/nnvm/top/nn.py", line 106, in compute_conv2d
    out = topi.nn.conv2d(inputs[0], inputs[1], strides, padding)
  File "<decorator-gen-27>", line 2, in conv2d
  File "/Users/imai/.pyenv/versions/3.5.3/lib/python3.5/site-packages/tvm-0.1.0-py3.5-macosx-10.10-x86_64.egg/tvm/target.py", line 217, in dispatch_func
    return func(*args, **kwargs)
  File "/Users/imai/.pyenv/versions/3.5.3/lib/python3.5/site-packages/topi-0.1.0-py3.5.egg/topi/nn/conv2d.py", line 82, in conv2d
    return conv2d_nchw(data, kernel, stride, padding)
  File "/Users/imai/.pyenv/versions/3.5.3/lib/python3.5/site-packages/topi-0.1.0-py3.5.egg/topi/nn/conv2d.py", line 260, in conv2d_nchw
    batch, in_channel, in_height, in_width = Input.shape
ValueError: not enough values to unpack (expected 4, got 0)
```

**Comment by masahi** [comment]
@imai-lm I reported the same error at

**Comment by tqchen** [comment]
cc @zhreshold

**Comment by zhreshold** [comment]
The onnx model from torchvision seems to have `expand` operator that don't appear in onnx/defs at all.
I am so sick about this.

**Comment by masahi** [comment]
@zhreshold Ah, that's bad. No idea how to get around this.

**Comment by tqchen** [comment]
@masahi can you put your model.proto on a gist or somewhere and provide a producible script to your problematic testcase reported in #202 ?

**Comment by masahi** [comment]
@tqchen I put imagenet.proto [here](https://drive.google.com/open?id=0Bwz2ceHveK8GaFpfM01iMnM4Ykk) on my google drive.

The test script is [here](https://gist.github.com/masahi/47fdf034a36cad282798c868667e9819)

**Comment by imai-lm** [comment]
@zhreshold I was also sick around that. See [here](https://github.com/pytorch/pytorch/issues/3257#issuecomment-340115211).

**Comment by masahi** [comment]
@imai-lm Oh...
This whole ONNX business is so unstable!

**Comment by tqchen** [comment]
can you check the reproducible case @masahi posted @zhreshold ?

**Comment by imai-lm** [comment]
I have tried the PR https://github.com/pytorch/pytorch/pull/3325 to the pytorch-AlexNet model, but I got `Expand` op instead of `expand`... It may take time until it becomes stable.

**Comment by imai-lm** [comment]
The problem above was resolved; the latest PyTorch can produce an ONNX-compliant model with `torch.onnx.export()`, while the API still has some bug in verbose mode (printed output contains `Expand`, but there are postprocesses that erase the ops).
@zhreshold Now you can try using PyTorch for ONNX model generation.

**Comment by imai-lm** [comment]
@tqchen The original problem is not solved yet... Why did you close this issue?

**Comment by tqchen** [comment]
ah sorry about that, I thought it is resolved

**Comment by bonotake** [comment]
@zhreshold How about this issue? Now could you confirm this can be reproduced on your environment?

**Comment by bonotake** [comment]
ahh, maybe I found the cause... It is just because I put a wrong shape_dict to nnvm.compiler.build()... I should have put `shape_dict = {'input_0': x.shape}` for example, but the key (the name of input layer) was wrong.

**Comment by masahi** [comment]
@bonotake for loading imagenet models, I recommend loading from gluon model zoo instead of pytorch.
It is more stable. I can already load resnet, vgg, etc.

**Comment by bonotake** [comment]
@masahi I need to use PyTorch models... ImageNet is not my primary purpose. I just want to try PyTorch-ONNX-NNVM path.

**Comment by masahi** [comment]
@bonotake loading from pytorch is a tough ride. so far, I was able to load [this](https://github.com/twtygqyy/pytorch-SRResNet) super res model, but that's it. 

Recently, conv2d_transpose was supported in nnvm. so I tried to load some pytorch models that use ConvTranspose2d, such as dcgan or fcn. But turned out pytorch exports nn.ConvTranspose2d as Caffe2ConvTranspose op in onnx! At that point, I gave up.
